### PR TITLE
feat(dev): direnv and vscode improvements

### PR DESCRIPTION
**Pull request by dcramer**
- Add support for nvm/node_modules via direnv
- Add defined tabsize for various languages
- Add support for python via direnv (virtualenvwrapper and local)

### CODE REVIEW

File: .envrc [hunk 1]
```diff
@@ -0,0 +1,125 @@
+set -e
+layout node
```
Comment (dcramer) [inline .envrc]: this makes node_modules/.bin exist on path

File: .envrc [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+}
+
+requires_nvm(){
+  __load_or_install_nvm
+  __nvm_use_or_install_version
+  __direnv_nvm_use_node
+  __npm_install_and_layout
+}
+
+requires_nvm
```
Comment (dcramer) [inline .envrc]: auto activates node version, and prompts/installs nvm if needed

File: .envrc [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    log_error "Install nvm first and make sure it is in your path and try again"
+    _log warn "To install NVM visit https://github.com/creationix/nvm#installation"
+    exit
+  fi
+}
+
+__npm_install_and_layout(){
+  if [ ! -d ./node_modules ]; then
+    # no node modules... run npm install
+    npm install
```
Comment (mattrobenolt) [inline .envrc]: Isn't this sorta what we don't wanna do? Shouldn't we be `yarn install`ing?

Also kinda weird IMO that we would `npm install` just by `cd` into a directory.

**Review by mattrobenolt — APPROVED** [review]

**Comment by dcramer** [comment]
Alreight after some more convo, I added python's virtualenv here as well.

1. It will look for $WORKON_HOME/sentry by default
2. If it can't find that, it will create a new environment
3. It **does not** call activate/deactivate, but only instantiates environments and adjusts PATh
4. You can override "sentry" using $SENTRY_ENV_NAME

**Comment by dcramer** [comment]
Alright so open to feedback on what kind of workflows this would break for people

**Comment by dcramer** [comment]
Ok so I'd love to get feedback on an improved concept around this:

- We'd create a new repository which is focused on "local development of sentry [in general]"
- The repository would contain the kinds of things we have in here, explicitly intended to setup a staging/non-vm environment
- The repository would be configurable to pull down the open source bits (sentry, sentry-plugins), but also our closed source bits
- All local dev configuration could then be easily shared, and done-once

**Comment by dcramer** [comment]
Ignoring my last comment, one last thought:

we could make this be .envrc.default, and have 'make' prompt you to copy a .envrc.sample to .envrc or similar

the .envrc.sample would just import defaults and "layout_sentry" or something similar

**Comment by EvanPurkhiser** [comment]
> we could make this be .envrc.default, and have 'make' prompt you to copy a .envrc.sample to .envrc or similar

This seems pretty reasonable

**Comment by joshuarli** [comment]
Closing; this was superseded by various direnv commits a while ago.

**Pull request closed without merging**